src/helpers/cardInfoUtil.js

```diff
@@ -10,20 +10,39 @@ const OPEN_RESULT_SETTING = <FormattedMessage id="cardInfo.opResultSet" defaultM
 const VOTING = <FormattedMessage id="cardInfo.vote" defaultMessage="Voting" />;
 const FINALIZING = <FormattedMessage id="cardInfo.final" defaultMessage="Finalizing" />;
 const WITHDRAWING = <FormattedMessage id="cardInfo.withdraw" defaultMessage="Withdraw" />;
-const BLOCK = 'Block:';
-const RANGE_SEPARATOR = 'to';
-const ANYTIME = 'anytime';
+
 
 const POS_TOPIC_CREATED = 0;
 const POS_BETTING = 1;
 const POS_ORACLE_RESULT_SETTING = 2;
 const POS_OPEN_RESULT_SETTING = 3;
 
+const messages = defineMessages({
+  block: {
+    id: 'str.block',
+    defaultMessage: 'Block',
+  },
+  to: {
+    id: 'cardInfo.to',
+    defaultMessage: 'To',
+  },
+  anytime: {
+    id: 'str.anytime',
+    defaultMessage: 'anytime',
+  },
+});
+
 class CardInfoUtil {
-  static getSteps(blockTime, cOracle, dOracles, isTopicDetail) {
+  static getSteps(blockTime, cOracle, dOracles, isTopicDetail, locale, localeMessages) {
     if (_.isUndefined(blockTime) && _.isUndefined(cOracle)) {
       return false;
     }
+    const intlProvider = new IntlProvider({ locale, messages: localeMessages }, {});
+    const { intl } = intlProvider.getChildContext();
```

Comment:
looks like this is redundant code in multiple places. make another helper utility method that gets the `intl` provider.

or better yet, make a utility method that will automatically get the `intl` var without having to pass in the `locale` and `localeMessages`.

follow a singleton pattern.